sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -729,6 +729,7 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
       def unapply(expr: Expression): Option[Attribute] = {
         expr match {
           case attr: Attribute => Some(attr)
+          case Cast(IntegralType(), StringType, _) => None
```

Comment:
good catch! I'm thinking if we should be more conservative here. How about
```
case Cast(child @ IntegralType(), dt: IntegralType, _) => if Cast.canUpCast...
```

Reply:
+1